cpp/tests/io/parquet_chunked_reader_test.cu

```diff
@@ -953,64 +953,283 @@ TEST_F(ParquetChunkedReaderTest, TestChunkedReadNullCount)
-  }
+void input_limit_test_write(std::string const& base_path, cudf::table_view const& t)
+{
+  // no compression
+  input_limit_test_write_one(base_path + "_a.parquet",
+                             t,
+                             cudf::io::compression_type::NONE,
+                             cudf::io::dictionary_policy::NEVER);
+  // compression with a codec that uses a lot of scratch space at decode time (2.5x the total
+  // decompressed buffer size)
+  input_limit_test_write_one(base_path + "_b.parquet",
+                             t,
+                             cudf::io::compression_type::ZSTD,
+                             cudf::io::dictionary_policy::NEVER);
+  // compression with a codec that uses no scratch space at decode time
+  input_limit_test_write_one(base_path + "_c.parquet",
+                             t,
+                             cudf::io::compression_type::SNAPPY,
+                             cudf::io::dictionary_policy::NEVER);
+  input_limit_test_write_one(base_path + "_d.parquet",
+                             t,
+                             cudf::io::compression_type::SNAPPY,
+                             cudf::io::dictionary_policy::ALWAYS);
+}
 
-  {
-    // 25 chunks of 100k rows each
-    auto const [result, num_chunks] = chunked_read(filepath, 0, 4000000);
-    EXPECT_EQ(num_chunks, 25);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->get_column(0));
-  }
+void input_limit_test_read(std::string const& base_path,
+                           cudf::table_view const& t,
+                           size_t output_limit,
+                           size_t input_limit,
+                           int const expected_chunk_counts[4])
+{
+  std::vector<std::string> file_suffix{"_a.parquet", "_b.parquet", "_c.parquet", "_d.parquet"};
+  CUDF_EXPECTS(file_suffix.size() == 4,
+               "Unexpected mismatch between number of test cases and result count");
 
-  {
-    // 25 chunks of 100k rows each
-    auto const [result, num_chunks] = chunked_read(filepath, 0, 4100000);
-    EXPECT_EQ(num_chunks, 25);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->get_column(0));
+  for (size_t idx = 0; idx < file_suffix.size(); idx++) {
+    auto result = chunked_read(base_path + file_suffix[idx], output_limit, input_limit);
+    CUDF_EXPECTS(result.second == expected_chunk_counts[idx],
+                 "Unexpected number of chunks produced in chunk read");
+    CUDF_TEST_EXPECT_TABLES_EQUIVALENT(*result.first, t);
   }
+}
 
-  {
-    // 12 chunks of 200k rows each, plus 1 final chunk of 100k rows.
-    auto const [result, num_chunks] = chunked_read(filepath, 0, 8002301);
-    EXPECT_EQ(num_chunks, 13);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->get_column(0));
-  }
+struct ParquetChunkedReaderInputLimitConstrainedTest : public cudf::test::BaseFixture {};
+
+TEST_F(ParquetChunkedReaderInputLimitConstrainedTest, SingleFixedWidthColumn)
```

Comment:
Supercalifragilisticexpialidocious has been real quiet since the new test name dropped :)